packages/rest/src/coercion/coerce-parameter.ts

```diff
@@ -169,10 +177,45 @@ function coerceObject(input: string | object, spec: ParameterObject) {
   if (typeof data !== 'object' || Array.isArray(data))
     throw RestHttpErrors.invalidData(input, spec.name);
 
-  // TODO(bajtos) apply coercion based on properties defined by spec.schema
+  const schema = extractSchemaFromSpec(spec);
+  if (schema) {
+    // Apply coercion based on properties defined by spec.schema
+    await validateValueAgainstSchema(
```

Comment:
`validateValueAgainstSchema` throws an error when AJV validation fails. Is it ok for `coerceObject` to throw validation errors? 

`validateValueAgainstSchema` prints debug logs referring to `request body`. If we are going to use `validateValueAgainstSchema` for non-body parameters (which consider as a good idea), then the debug logs need to be updated accordingly.

We should also take a look at the validation errors produced for non-body parameters and make sure they provide enough information about which parameter failed the validation.

Another aspect to consider: should we report the first validation error only (I think that's the current design)? IMO, it's easier for users when all errors are reported at once. A possible solution for this is to build a "composite" object from all parameters (e.g. `{filter: <value from query string>, authorization: <value from headers>}`), a "composite" schema from Operation Object, and then call AJV only once. I think this way the error messages will include the parameter name ("filter", "authorization") as part of the description (path to the property with a wrong value).

Reply:
Having an aggregated schema that stitch all incoming params (including the body) is an interesting idea that we should explore. But let's leave it out of scope for this PR.

Reply:
> Having an aggregated schema that stitches all incoming params (including the body) is an interesting idea that we should explore.

@bajtos @raymondfeng Thanks for suggestion, I also tend to create a separate story for it. In this PR, I turned `validateValueAgainstSchema` a generic function that validates any value:

- rename 1st parameter: `data` --> `value`
- `ValueValidationOptions` extends `RequestBodyValidationOptions` with one more property `position`
- throws invalid request body error when `position` is 'body', throws invalid data error otherwise.

The new story can create a new rest http error for other invalid data. 
WDYT?

Reply:
creating the follow-up story, post here soon.